### ISSUE TITLE
Reload config on SIGHUP

### ISSUE THREAD

**Issue by kderme** [input-output-hk/cardano-db-sync#1201]
db-sync should reload the configuration in case of a SIGHUP. This requires proper handlers and can be related to https://github.com/input-output-hk/cardano-db-sync/issues/1010.

**Comment by erikd** [comment]
What problem does this solve? How often does the config change and need reloading?

**Comment by nrdxp** [comment]
Just to add a reference point, this would make my work here a little cleaner:
https://github.com/input-output-hk/cardano-world/pull/70

Basically what I am doing here is using dynamic db credentials generated by Vault's db secrets engine. Those credentials can be extended up to a certain maximum lease time. Once the max lease time has passed, the entrypoint kills db-sync (which is has running as a background job) grabs fresh login information from Vault again, and then restarts db-sync.

A similar process occurs when the Vault token itself expires and Nomad restarts the entire task. If I could send SIGHUP in both cases, it would be just a bit nicer, since there would be no service interuption. But with that said, the interuption is fairly minimal as is (only a second or 2).

**Comment by rdlrt** [comment]
> Basically what I am doing here is using dynamic db credentials generated by Vault's db secrets engine.

Isn't it more secure/better practice to use ident/socket instead of creds for DB? You can have alternate users authenticate with creds as desired.

**Comment by nrdxp** [comment]
@rdlrt you are correct there is a better way of doing it that that I just become aware of. I dunno what you mean by "forced usage" necessarily since there is an officially supported [backend](https://developer.hashicorp.com/vault/tutorials/db-credentials/database-secrets) expressly for this purpose. However, I did not realize that with the proper creation and revocation SQL statements it is possible to switch to a permanent role, from which these dynamic roles inherit their permissions, immediately after login.

This has some nice properties, for example new tables are owned by the permanent user (whose credentials are never exposed) so there are no access issues once the dynamic user is destroyed, but even more importantly I can completely drop the dynamic user after its Vault lease expires, but the open connection continues to function as the permanent role until the job stops, negating the need for application restarts at all.

**Comment by rdlrt** [comment]
> I dunno what you mean by "forced usage"necessarily since there is an officially supported [backend](https://developer.hashicorp.com/vault/tutorials/db-credentials/database-secrets) expressly for this purpose.

I meant use of credentials for dbsync is less than ideal IMO, keeping the scope to dbsync itself - where the actual dbsync database and postgres are on same machine and the user used to connect from dbsync to postgres:
- DBSync itself can connect to postgres [using socket file](https://github.com/input-output-hk/cardano-db-sync/blob/2b64878dab5165766730372e2ed3e47bae918360/config/pgpass-mainnet#L1) (thus, use of vault becomes irrelevant for dbsync itself - as there are no credentials needed), this is true even for docker.
- For alternate user (for consumption of database) - sure , you can have credential based user who creds are in vault - but that becomes a downstream application operations, thus - splits any concerns regarding operations of dbsync itself.

Thus, while that backend might be useful for consumption use-cases, it isnt for the connectivity between dbsync and postgres database itself

**Comment by nrdxp** [comment]
The postgres database is not on the same machine in our case.

**Comment by nrdxp** [comment]
Yeah, well this isn't how it was handed off to me to begin with. Initially we had our own patroni managed db instance in cluster that db-sync targeted. It wasn't on the same machine either, to alleviate myself of the maintenance burden I moved it to an RDS cluster. So far I haven't seen any issues, but if it becomes an issue at some point in the future I can just move it again.

**Comment by erikd** [comment]
There are two potential issues:

* Syncing from genesis will take 4 times (or more) longer than it should.
* As blocks sizes and transactions per block increases, it may reach a point where new blocks arriving cannot be inserted in the database in time enough to keep up with newly arriving blocks.
